jetstream_consumer.go

```diff
@@ -0,0 +1,335 @@
+	switch p {
+	case ReplayOriginal:
+		return json.Marshal("original")
+	case ReplayInstant:
+		return json.Marshal("instant")
+	default:
+		return nil, fmt.Errorf("unknown replay policy %v", p)
+	}
+}
+
+var (
+	AckAck      = []byte("+ACK")
+	AckNak      = []byte("-NAK")
+	AckProgress = []byte("+WPI")
+	AckNext     = []byte("+NXT")
+	AckTerm     = []byte("+TERM")
+)
+
+type DeliverPolicy int
+
+const (
+	DeliverAll DeliverPolicy = iota
+	DeliverLast
+	DeliverNew
+	DeliverByStartSequence
+	DeliverByStartTime
+)
+
+func (p DeliverPolicy) String() string {
+	switch p {
+	case DeliverAll:
+		return "All"
+	case DeliverLast:
+		return "Last"
+	case DeliverNew:
+		return "New"
+	case DeliverByStartSequence:
+		return "By Start Sequence"
+	case DeliverByStartTime:
+		return "By Start Time"
+	default:
+		return "Unknown Deliver Policy"
+	}
+}
+
+func (p *DeliverPolicy) UnmarshalJSON(data []byte) error {
+	switch string(data) {
+	case jsonString("all"), jsonString("undefined"):
+		*p = DeliverAll
+	case jsonString("last"):
+		*p = DeliverLast
+	case jsonString("new"):
+		*p = DeliverNew
+	case jsonString("by_start_sequence"):
+		*p = DeliverByStartSequence
+	case jsonString("by_start_time"):
+		*p = DeliverByStartTime
+	}
+
+	return nil
+}
+
+func (p DeliverPolicy) MarshalJSON() ([]byte, error) {
+	switch p {
+	case DeliverAll:
+		return json.Marshal("all")
+	case DeliverLast:
+		return json.Marshal("last")
+	case DeliverNew:
+		return json.Marshal("new")
+	case DeliverByStartSequence:
+		return json.Marshal("by_start_sequence")
+	case DeliverByStartTime:
+		return json.Marshal("by_start_time")
+	default:
+		return nil, fmt.Errorf("unknown deliver policy %v", p)
+	}
+}
+
+// ConsumerConfig is the configuration for a JetStream consumes
+//
+// NATS Schema Type io.nats.jetstream.api.v1.consumer_configuration
+type ConsumerConfig struct {
+	Durable         string        `json:"durable_name,omitempty"`
+	DeliverSubject  string        `json:"deliver_subject,omitempty"`
+	DeliverPolicy   DeliverPolicy `json:"deliver_policy"`
+	OptStartSeq     uint64        `json:"opt_start_seq,omitempty"`
+	OptStartTime    *time.Time    `json:"opt_start_time,omitempty"`
+	AckPolicy       AckPolicy     `json:"ack_policy"`
+	AckWait         time.Duration `json:"ack_wait,omitempty"`
+	MaxDeliver      int           `json:"max_deliver,omitempty"`
+	FilterSubject   string        `json:"filter_subject,omitempty"`
+	ReplayPolicy    ReplayPolicy  `json:"replay_policy"`
+	SampleFrequency string        `json:"sample_freq,omitempty"`
+	RateLimit       uint64        `json:"rate_limit_bps,omitempty"`
+}
+
+type SequencePair struct {
+	ConsumerSeq uint64 `json:"consumer_seq"`
+	StreamSeq   uint64 `json:"stream_seq"`
+}
+
+type ConsumerInfo struct {
+	Stream         string         `json:"stream_name"`
+	Name           string         `json:"name"`
+	Config         ConsumerConfig `json:"config"`
+	Created        time.Time      `json:"created"`
```

Comment:
Same, here we have a very large number that is not friendly to anything other than Go.

Reply:
Go times in json is a string in rfc3339 format not a big number

Reply:
That is possibly even worse, depending on the language difficult to parse. Unix time is always super easy.